### PR TITLE
saltnado bugfixes for ldap & syndics

### PR DESCRIPTION
**Pull request by mattp-**
### What does this PR do?
fix syndic_wait handling in saltnado in syndic topologies:

 as is, yielding sleep before registering the event listener as actually causing saltnado to _miss_ the event entirely. instead of waiting before registering  instead we just add the syndic_wait to the futures base timeout. This issue only affected synchronous localclient calls

also included, a fix for ldap group based perms that mimics cherrypy compatibility.

### New Behavior
- saltnado now works with syndic topologies
- saltnado now works with group perms with ldap

### Tests written?
Yes

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/netapi/rest_tornado/saltnado.py [hunk 1]
```diff
@@ -740,7 +740,17 @@ def post(self):
...
+            # Get sum of '*' perms, user-specific perms, and group-specific perms
+            perms = eauth.get(token['name'], [])
+            perms.extend(eauth.get('*', []))
+
+            if 'groups' in token and token['groups']:
+                user_groups = set(token['groups'])
+                eauth_groups = set([i.rstrip('%') for i in eauth.keys() if i.endswith('%')])
+
+                for group in user_groups & eauth_groups:
+                    perms.extend(eauth['{0}%'.format(group)])
```
Comment (DmitryKuzmenko) [inline salt/netapi/rest_tornado/saltnado.py]: I see a behavior change here. If not user name nor groups are in eauth list before we were getting `send_error(401)`. Now we're returning with empty perms list.
If this change is expected I'm OK. If not it's better to detect match with a flag keeping `send_error` if not name nor group match found.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
@mattp-  could you please take a look to my inline comment?

File: salt/netapi/rest_tornado/saltnado.py (hunk 1 above)
Comment (mattp-) [reply salt/netapi/rest_tornado/saltnado.py]: good catch, i just copied this from rest_cherrypy. I'll restore send_error(401)

Comment (mattp-) [reply salt/netapi/rest_tornado/saltnado.py]: actually on further review it looks like this is expected behavior in cherrypy. I am indifferent here, is there any policy of attempting to make saltnado compatible with cherrypy? is cherrypy the canonical implementation?

Comment (mattp-) [reply salt/netapi/rest_tornado/saltnado.py]: on further thought, I think aligning with cherrypy makes sense here. the reason I hit this, and likely why anyone else would, is theyve transitioned from cherrypy with pepper and are moving to tornado. functionality should be 1 to 1 as close as possible. so working as designed :)

Comment (DmitryKuzmenko) [reply salt/netapi/rest_tornado/saltnado.py]: I'm OK then. Thank you for your work.

**Comment by DmitryKuzmenko** [comment]
re-run all

**Review by DmitryKuzmenko — APPROVED** [review]

**Review by skizunov — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
We hit a test failure here: https://jenkins.saltstack.com/job/PR/job/salt-pr-rs-cent7-n/17790/testReport/junit/integration.netapi.rest_tornado.test_app/TestSaltAPIHandler/test_simple_local_post_only_dictionary_request_with_order_masters/

**Comment by mattp-** [comment]
rerun all

**Comment by mattp-** [comment]
rerun all

**Comment by cachedout** [comment]
re-run all

**Comment by rallytime** [comment]
@mattp- This is causing some related tests to fail: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu16-py3/8619/

**Pull request merged**